src/shared/slideRegistry.ts

```diff
@@ -0,0 +1,92 @@
+import { ComponentInternalInstance, shallowReactive } from 'vue'
+
+type SlideEventName = 'register' | 'unregister'
+
+interface SlideRegistrationEvent {
+  slide: ComponentInternalInstance
+  index: number
+}
+
+type SlideRegistrationCallback = (event: SlideRegistrationEvent) => void
+
+const createSlideRegistry = () => {
+  const slides = shallowReactive<Array<ComponentInternalInstance>>([])
+  const listeners: Record<SlideEventName, Set<SlideRegistrationCallback>> = {
```

Comment:
Instead of using a custom listener implementation here, you could pass `emit` as an argument to the `createSlideRegistry`